### ISSUE TITLE
last_updated field in addons collection does not reflect the modification date of the add-on file

### ISSUE THREAD

**Issue by Rob--W** [mozilla/addons-server#14209]
### Describe the problem and steps to reproduce it:
Load https://addons.mozilla.org/api/v4/accounts/account/mozilla/collections/7e8d6dc651b54ab385fb8791bf9dac/addons/ and look for `last_updated` (at uBlock Origin). This [is used](https://github.com/mozilla-mobile/android-components/blob/6ae25079c9e5386353ddcf6b0810a34be9dd2649/components/feature/addons/src/main/java/mozilla/components/feature/addons/amo/AddonCollectionProvider.kt#L29-L83) to populate the content of the addons detail view in Fenix (I tested on Firefox Beta).

### What happened?
```
        "last_updated": "2020-05-07T00:40:10Z",
```

### What did you expect to happen?
```
        "last_updated": "2020-04-16T21:03:41Z"`,
```

### Anything else we should know?
The date in my expectation comes from the `.results[0].files[0].created_date` field at https://addons.mozilla.org/api/v4/addons/addon/ublock-origin/versions/?page_size=1&lang=en-US (
`"files":[{"id":3551054,"created":"2020-04-16T21:03:41Z"`) . This date is also shown when I look at the most recent version of uBlock Origin (1.26.2) in the user-facing interface at https://addons.mozilla.org/en-US/firefox/addon/ublock-origin/versions/

**Comment by wagnerand** [comment]
This looks like a duplicate/consequence of https://github.com/mozilla/addons-server/issues/6929.

**Comment by eviljeff** [comment]
yeah, the last_updated field will be different if there are unlisted versions added in the meantime.  It will also be different if the add-on metadata was edited via devhub; or if a file for a listed version was uploaded and then deleted/rejected. 

tl;dr `last_updated` in the addon detail response != the date of the the latest version/file uploaded.

**Comment by Rob--W** [comment]
So is that a bug in the client then?

What should clients use in order to determine whether the add-on package has been updated? From the user's / client's perspective, I expect the update date to reflect the last update date of the add-on (and that it also matches what AMO shows as the update date).

**Comment by eviljeff** [comment]
> So is that a bug in the client then?

yes.

> What should clients use in order to determine whether the add-on package has been updated? 

assuming the add-on package is the listed version on AMO, then the data in `current_version`

> From the user's / client's perspective, I expect the update date to reflect the last update date of the add-on

It does, just not always the date the most recent publicly available listed version was created.

> (and that it also matches what AMO shows as the update date).

from the issue @wagnerand  referenced, apparently that was fixed in mozilla/addons-frontend#8861

**Comment by Rob--W** [comment]
Thanks @eviljeff . I will move this issue to A-C (https://github.com/mozilla-mobile/android-components/issues/6935).
